spec/unit/controllers/services/service_instances_controller_spec.rb

```diff
@@ -4026,20 +4054,13 @@ def verify_forbidden(user)
 
               it "has a #{expected_status} http status code" do
                 get "/v2/service_instances/#{instance.guid}/shared_to"
-                expect(last_response.status).to eq(expected_status), "Expected #{expected_status}, got: #{last_response.status}, role: #{role}"
+                expect(last_response.status).to eq(expected_status), "Expected #{expected_status}, got: #{last_response.status}, role: #{role}, response: #{last_response.body}"
               end
             end
           end
         end
 
         context 'when the user is a member of the org/space where the service instance was shared to' do
```

Comment:
We'd expect to see a test case for the encountered bug, e.g. given two developers: a developer in `space-1` and `space-2` and a developer in `space-1`:
* the first developer shares service instance `SI` into `space-2`
* the second developer can unshare `SI` out of `space-2`

Reply:
@elenasharma we're not clear what you mean here. This endpoint is not for unsharing, it's for viewing the space GUIDs where a service instance has been shared to. Developers in recipient spaces shouldn't have access to this, because then they'd see GUIDs for other spaces where they shouldn't normally have access.

We do test this workflow in the v3/controllers unit test for unsharing: https://github.com/cloudfoundry/cloud_controller_ng/blob/master/spec/unit/controllers/v3/service_instances_controller_spec.rb#L480-L505

If we've misunderstood what you meant, please let us know!